lib/linguist/heuristics.rb

```diff
@@ -541,5 +541,13 @@ def call(data)
       end
     end
 
+    disambiguate ".yy" do |data|
+      Language["JSON"] if /\"modelName\"\:\s?\"/.match(data)
+    end
```

Comment:
We'll need to add a case for Yacc here.

Reply:
Right, my *need* above might be a bit strong. It's *better* if we have a case for Yacc :smile: 
Is the presence of the `modelName` key sufficient to identify GML JSON files? (i.e., is it mandatory in those files?)

Reply:
Pretty much. All the .yy files I've seen have a modelName. As does the .yyp file. But like you said - the .yyp file isn't documented in the languages.rb file so I've removed it in the latest commit. I've extended this testcase to first go for XML, then Graph Markup Language, and then a GameMaker file. The latter is harder because its a generic programming language and has no "required" keywords in the file. XML does, and it looks like Graph Markup Language does too (http://www.fim.uni-passau.de/fileadmin/files/lehrstuhl/brandenburg/projekte/gml/gml-technical-report.pdf) - the ``graph [`` and potentially the ``node [``.